packages/xo-web/src/xo-app/host/tab-advanced.js

```diff
@@ -279,7 +282,9 @@ export default class extends Component {
                 <tr>
                   <th>{_('hyperThreading')}</th>
                   <td>
-                    {this._isHtEnabled()
+                    {isHtEnabled === null
+                      ? _('hyperThreadingNotAvailable')
```

Comment:
What if it's a XCP-ng host but without the plugin installed? It's possible, right?

Reply:
https://github.com/vatesfr/xen-orchestra/pull/4285#discussion_r295328249

https://github.com/vatesfr/xen-orchestra/pull/4285/files#diff-0d7c8183f21c8290ffbeb18bca5c0f88R2365

Reply:
Yes but if it's a XCP-ng host without the plugin installed, it will throw on the server side, and we'll tell the user that the detection is only available on XCP-ng, even though he *is* on XCP-ng. Or did I miss something?

Reply:
does `error.code === 'XENAPI_MISSING_PLUGIN'` is only for `xenServer` ?

Reply:
```js
async isHyperThreading(hostId) {
    try {
      return (
        (await this.call(
          'host.call_plugin',
          this.getObject(hostId).$ref,
          'hyperthreading.py',
          'get_hyperthreading',
          {}
        )) !== 'false'
      )
    } catch (error) {
      if (
        error.code === 'XENAPI_MISSING_PLUGIN' ||
        error.code === 'UNKNOWN_XENAPI_PLUGIN_FUNCTION'
      ) {
        return null
      } else {
        throw error
      }
    }
  }
```
If the plugin is not installed, this function will return `null` which is intepreted front side as a missing plugin.

```js
{isHtEnabled === null
                      ? _('hyperThreadingNotAvailable')
```

Reply:
I can't be sure about that but I don't know why it would be any different on XCP-ng.

Reply:
@julien-f is `XENAPI_MISSING_PLUGIN` error is specified to `xenServer` or it's also for `xcp-ng`

Reply:
Should be the same.

Reply:
@HamadaBrest So you can handle this by either handling 2 "error" cases ("not available on XS" & "plugin not installed on XCP-ng") or keep it simple and simply tell the user that Hyper-threading detection is not available.